### PR TITLE
array accessors

### LINKED ISSUE #578: Array access proposal
Current behavior:
  - SVD array has sequential indexes started from 0 and sequential addresses => create Rust array and add it to Cluster/RegisterBlock
  - otherwise create Rust instance for each element of SVD array

Proposal (breaking):
  - For each SVD array with sequential addresses create Rust array. Also create functions with appropriate names to access particular cluster/register in array.
 
So for example instead of:
```rust
struct RegisterBlock {
   ccr1: CCR,
   ccr2: CCR,
   ccr3: CCR,
   ccr4: CCR,
}
```
we will have:
```rust
struct RegisterBlock {
   ccr: [CCR; 4],
}
impl RegisterBlock {
  pub fn ccr1() -> &CCR {
    &self.ccr[0]
  }
  ...
}
```

Or instead of:
```rust
struct RegisterBlock {
   cha: CH,
   chb: CH,
}
```
we will have:
```rust
struct RegisterBlock {
   ch: [CH; 2],
}
impl RegisterBlock {
  pub fn cha() -> &CH {
    &self.ch[0]
  }
  pub fn chb() -> &CH {
    &self.ch[1]
  }
}
```

cc @adamgreig  @therealprof

### PR DESCRIPTION
**Pull request by burrbull**
Closes #578

### CODE REVIEW

**Comment by burrbull** [comment]
@adamgreig  @therealprof  ping

**Comment by Emilgardis** [comment]
I can't get this to work, do you have an example svd to demonstrate with?

**Comment by burrbull** [comment]
> I can't get this to work, do you have an example svd to demonstrate with?

Any SVD from https://stm32-rs.github.io/stm32-rs/

**Review by Emilgardis — DISMISSED** [review]
bors r+

**Comment by Emilgardis** [comment]
SiliconLabs svds has moved to subfolders

https://github.com/posborne/cmsis-svd/blob/master/data/SiliconLabs/SiM3_NRND/SIM3C1x4.svd for example

File: ci/script.sh [hunk 1]
```diff
@@ -531,15 +531,15 @@ main() {
...
+            test_svd_for_target cortex-m https://raw.githubusercontent.com/posborne/cmsis-svd/master/data/SiliconLabs/SiM3_NRND/SIM3C1x6_SVD
+            test_svd_for_target cortex-m https://raw.githubusercontent.com/posborne/cmsis-svd/master/data/SiliconLabs/SiM3_NRND/SIM3C1x7_SVD
+            test_svd_for_target cortex-m https://raw.githubusercontent.com/posborne/cmsis-svd/master/data/SiliconLabs/SiM3_NRND/SIM3L1x4_SVD
+            test_svd_for_target cortex-m https://raw.githubusercontent.com/posborne/cmsis-svd/master/data/SiliconLabs/SiM3_NRND/SIM3L1x6_SVD
+            test_svd_for_target cortex-m https://raw.githubusercontent.com/posborne/cmsis-svd/master/data/SiliconLabs/SiM3_NRND/SIM3L1x7_SVD
+            test_svd_for_target cortex-m https://raw.githubusercontent.com/posborne/cmsis-svd/master/data/SiliconLabs/SiM3_NRND/SIM3U1x4_SVD
+            test_svd_for_target cortex-m https://raw.githubusercontent.com/posborne/cmsis-svd/master/data/SiliconLabs/SiM3_NRND/SIM3U1x6_SVD
+            test_svd_for_target cortex-m https://raw.githubusercontent.com/posborne/cmsis-svd/master/data/SiliconLabs/SiM3_NRND/SIM3U1x7_SVD
```
Comment (Emilgardis) [inline ci/script.sh:542]: wouldn't 

```suggestion
            test_svd SiM3_NRND/SIM3C1x4
            test_svd SiM3_NRND/SIM3C1x6
            test_svd SiM3_NRND/SIM3C1x7
            test_svd SiM3_NRND/SIM3L1x4
            test_svd SiM3_NRND/SIM3L1x6
            test_svd SiM3_NRND/SIM3L1x7
            test_svd SiM3_NRND/SIM3U1x4
            test_svd SiM3_NRND/SIM3U1x6
            test_svd SiM3_NRND/SIM3U1x7
```

be enough?

Comment (burrbull) [reply ci/script.sh:542]: No. It was my initial idea.

**Review by Emilgardis — APPROVED** [review]
bors r+

**Review by Emilgardis — APPROVED** [review]
bors r+

**Pull request merged**